fairworkflows/fairworkflow.py

```diff
@@ -33,6 +35,78 @@ def __init__(self, description, uri=None):
         self._steps = {}
         self._last_step_added = None
 
+    @classmethod
+    def from_rdf(cls, rdf: rdflib.Graph, uri: str = None,
+                 fetch_steps: bool = False):
+        """Construct Fair Workflow from existing RDF.
+
+        Args:
+            rdf: RDF graph containing information about the workflow and
+                possibly it's associated steps. Should use plex ontology.
+            uri: URI of the workflow
+            fetch_steps: toggles fetching steps. I.e. if we encounter steps
+                that are part of the workflow, but are not specified in the
+                RDF we try fetching them from nanopub
+        """
+        rdf = copy(rdf)  # Make sure we don't mutate user RDF
+        if rdflib.URIRef(uri) not in rdf.subjects():
+            warnings.warn(f"Warning: Provided URI '{uri}' does not "
+                          f"match any subject in provided rdf graph.")
+        self = cls(uri=uri)
+        self._extract_steps(rdf, uri, fetch_steps)
+        self._rdf = rdf
+        self.anonymise_rdf()
+        return self
+
+    def _extract_steps(self, rdf, uri, fetch_steps=False):
+        """Extract FairStep objects from rdf.
+
+        Create FairStep objects for all steps in the passed RDF. Removes
+        triples describing steps from rdf, those will be represented in
+        the separate step RDF. Optionally try to fetch steps from nanopub.
+        """
+        step_refs = rdf.subjects(predicate=Nanopub.PPLAN.isStepOfPlan,
+                                 object=rdflib.URIRef(uri))
+        for step_ref in step_refs:
+            step_uri = str(step_ref)
+            step = self._extract_step_from_rdf(step_uri, rdf)
+            if step is None and fetch_steps:
+                step = self._fetch_step(uri=step_uri)
+            if step is None:
+                warnings.warn(f'Could not get detailed information for '
+                              f'step {step_uri}, adding a FairStep '
+                              f'without attributes. This will limit '
+                              f'functionality of the FairWorkflow object.')
+                step = FairStep(uri=step_uri)
+            self._add_step(step)
+
+    @staticmethod
+    def _extract_step_from_rdf(uri, rdf: rdflib.Graph()) -> Optional[FairStep]:
+        step_rdf = rdflib.Graph()
+        for s, p, o in rdf.triples((rdflib.URIRef(uri), None, None)):
+            if p in FAIRSTEP_PREDICATES:
+                step_rdf.add((s, p, o))
+                rdf.remove((s, p, o))
+
+        if len(step_rdf) > 0:
+            return FairStep.from_rdf(step_rdf, uri=uri)
+        else:
+            return None
+
+    @staticmethod
+    def _fetch_step(uri: str) -> Optional[FairStep]:
+        try:
```

Comment:
Yes, I think this is the right way to do this for now. We will worry about adding other sources when we are given a clear example/use case.

Reply:
Bueno! :)